File: go/exercise14/tax_calculator.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	const tax float64 = 5.5

	scanner := bufio.NewScanner(os.Stdin)

	fmt.Print("What is the order amount? ")
	scanner.Scan()
	amount := scanner.Text()

	fmt.Print("What is the state? ")
	scanner.Scan()
	state := scanner.Text()

	subtotal, _ := strconv.ParseFloat(amount, 64)

	// Check the amount and state condition
	if state == "WI" {
		fmt.Printf("The subtotal is $%.2f\n", subtotal)
		fmt.Printf("The tax is $%.2f\n", (subtotal*tax)/100)
		fmt.Printf("The total is $%.2f\n", subtotal+((subtotal*tax)/100))
		return
	}

	fmt.Printf("The total is $%.2f\n", subtotal)
}

File: javascript/exercise14/tax-calculator.js
import { input } from "pynput";

const tax = 5.5;

const amount = Number(await input("What is the order amount? "));
const state = await input("What is the state? ");

// Check the amount and state condition
if (state.toUpperCase() === "WI") {
	console.log(`The subtotal is $${amount.toFixed(2)}`);
	console.log(`The tax is $${((amount * tax) / 100).toFixed(2)}`);
	console.log(`The total is $${(amount + (amount * tax) / 100).toFixed(2)}`);
	process.exit();
}

console.log(`The total is $${amount.toFixed(2)}`);
